manager/state/raft/raft_test.go

```diff
@@ -568,23 +571,39 @@ func TestRaftUnreachableNode(t *testing.T) {
 
 	ctx := context.Background()
 	// Add a new node, but don't start its server yet
-	n := raftutils.NewNode(t, clockSource, tc, raft.NewNodeOptions{JoinAddr: nodes[1].Address})
-	go n.Run(ctx)
+	nodes[2] = raftutils.NewNode(t, clockSource, tc, raft.NewNodeOptions{JoinAddr: nodes[1].Address})
+
+	err := nodes[2].JoinAndStart()
+	require.NoError(t, err, "can't join cluster")
+
+	go nodes[2].Run(ctx)
+
+	// Stop the Raft server of second node on purpose after joining
+	nodes[2].Server.Stop()
+	nodes[2].Listener.CloseListener()
 
 	raftutils.AdvanceTicks(clockSource, 5)
 	time.Sleep(100 * time.Millisecond)
 
-	raft.Register(n.Server, n.Node)
+	l, err := net.Listen("tcp", nodes[2].Opts.Addr)
```

Comment:
I think this can be `nodes[2].Address`. Then `Opts` doesn't need to be exported.